src/primitives/transaction.cpp

```diff
@@ -161,18 +160,26 @@ std::vector<char> loadTransaction(const std::vector<CMFToken> &tokens, std::vect
             seenCoinbaseMessage = true;
             break;
         }
-        case Consensus::TxInScript: {
-            if (inputs.empty()) throw std::runtime_error("TxInScript before TxInPrevHash");
-            if (inputCount == 0) { // copy all of the input tags
+        case Consensus::TxInputStackItem:
+        case Consensus::TxInputStackItemContinued: {
+            if (signatureCount == -1) { // copy all of the input tags
                 CDataStream stream(0, 4);
                 ser_writedata32(stream, nVersion);
                 for (unsigned int i = 0; i < index; ++i) {
                     tokens[i].Serialize(stream, 0, 4);
                 }
                 txData = std::vector<char>(stream.begin(), stream.end());
             }
+            if (signatureCount < 0 || token.tag == Consensus::TxInputStackItem)
+                signatureCount++;
+            if (static_cast<int>(inputs.size()) <= signatureCount)
+                throw std::runtime_error("TxInputStackItem* before TxInPrevHash");
+
             auto data = token.unsignedByteArray();
-            inputs[inputCount++].scriptSig = CScript(data.begin(), data.end());
+            if (data.size() == 1)
+                inputs[signatureCount].scriptSig << data.at(0);
+            else
+                inputs[signatureCount].scriptSig << data;
```

Comment:
I guess I should add a comment...

The wonderful CScript class has a really stupid design in its append method, if you append a vector, it prepends that with an extra byte for the push so as to keep a valid script.
So if I didn't do the first case, it would instead append 2 bytes, not 1.